src/components/textfieldControl/js/TextfieldSlider.js

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2013-2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /********************
+     * Textfield Slider *
+     ********************/
+
+    fluid.defaults("fluid.textfieldSlider", {
+        gradeNames: ["fluid.viewComponent"],
+        components: {
+            textfield: {
+                type: "fluid.textfield",
+                container: "{that}.dom.textfield",
+                options: {
+                    gradeNames: ["fluid.textfield.rangeController"],
+                    components: {
+                        controller: {
+                            options: {
+                                model: "{textfieldSlider}.model"
+                            }
+                        }
+                    },
+                    ariaOptions: "{textfieldSlider}.options.ariaOptions",
+                    strings: "{textfieldSlider}.options.strings"
+                }
+            },
+            slider: {
+                type: "fluid.slider",
+                container: "{textfieldSlider}.dom.slider",
+                options: {
+                    model: "{textfieldSlider}.model",
+                    ariaOptions: "{textfieldSlider}.options.ariaOptions",
+                    strings: "{textfieldSlider}.options.strings"
+                }
+            }
+        },
+        selectors: {
+            textfield: ".flc-textfieldSlider-field",
+            slider: ".flc-textfieldSlider-slider"
+        },
+        styles: {
+            container: "fl-textfieldSlider fl-focus"
+        },
+        model: {
+            value: null,
+            step: 1.0,
+            range: {
+                min: 0,
+                max: 100
+            }
+        },
+        modelRelay: {
+            target: "value",
+            singleTransform: {
+                type: "fluid.transforms.limitRange",
+                input: "{that}.model.value",
+                min: "{that}.options.range.min",
+                max: "{that}.options.range.max"
+            }
+        },
+
+        ariaOptions: {
+            // Specified by implementor
+            // ID of an external label to refer to with aria-labelledby
+            // attribute
+            // "aria-labelledby": ""
+        },
+        strings: {
+            // Specified by implementor
+            // text of label to apply to both textfield and slider input
+            // via aria-label attribute
+            // "aria-label": ""
```

Comment:
Using the name of `aria-label` for the string label of the text field slider might be hard for people who are not familiar with ARIA to understand. Probably use a non-technical name such as `textfieldSliderLabel`?